### PR TITLE
BAC 548 - Enabling Background Event Notification

### PR DESCRIPTION
**Pull request by thertzelle**
### What

This is a PR that supports adding a setNotify, effectively a listener, that will notify any clients when a advertising service characteristic is found after, this event only occurs during a disconnected state on a bonded connection.

Calls the delegate to notify a parent application that they might want to do something.

### Why

So that we know when we should ask for new data in a background situation.

### CODE REVIEW

File: BacTrackAPI.m [hunk 1]
```diff
@@ -152,11 +152,9 @@ -(id)init
...
        
         [self loadProtocolSelectors];
-        connectedPeripherals = [[NSMutableDictionary alloc] initWithCapacity:1];
-        connectedAPIEndpoints = [[NSMutableDictionary alloc] initWithCapacity:1];
-        cmanager = [[CBCentralManager alloc] initWithDelegate:self
-                                                           queue:nil
-                                                         options:[NSDictionary dictionaryWithObject:[NSNumber numberWithInt:0] forKey:CBCentralManagerOptionShowPowerAlertKey]];
+        connectedPeripherals = [[NSMutableDictionary alloc] initWithCapacity: 1];
+        connectedAPIEndpoints = [[NSMutableDictionary alloc] initWithCapacity: 1];
+        cmanager = [[CBCentralManager alloc] initWithDelegate: self queue: nil options: @{ CBCentralManagerOptionRestoreIdentifierKey: @"com.bactrack.centralmanager", CBCentralManagerOptionShowPowerAlertKey: @(NO)}];
```
Comment (thertzelle) [inline BacTrackAPI.m:157]: This is just a formatting situation and updating the NSDictionary to the modern short form and adding a required restoration identifier.

File: BacTrackAPI.m [hunk 2]
```diff
@@ -632,6 +630,9 @@ -(void)connectTimeout
 /*			     CBCentralManagerDelegate protocol methods beneeth here     */
 /****************************************************************************/
 
+- (void)centralManager:(CBCentralManager *)central willRestoreState:(NSDictionary<NSString *,id> *)dict {
+}
+
```
Comment (thertzelle) [inline BacTrackAPI.m:635]: Required delegate, but we dont need an implementation right now.

**Comment by samcdavid** [comment]
Is there any way these changes could affect the View app (taking a test with a breathalyzer) if the pod is updated?

**Review by jlstr — APPROVED** [review]

File: BacTrackAPI.m (hunk 2 above)
Comment (jlstr) [reply BacTrackAPI.m:635]: Does a syntax  to make an optional method in a protocol in Obj-C exist? just curious

**Review by samcdavid — APPROVED** [review]

**Comment by thertzelle** [comment]
> Is there any way these changes could affect the View app (taking a test with a breathalyzer) if the pod is updated?

Taken care of during stand up...No impact from these things from the extra dictionary reform and the additional functions.

File: BacTrackAPI.m (hunk 2 above)
Comment (thertzelle) [reply BacTrackAPI.m:635]: Something like:
```
@optional
- (void)BacTrackSkynSyncRequest;
```
written in the protocol, is what you would want to do. I would have done the same for this PR, but I was following the paradigm of no-optional delegates.
`willRestoreState` is required, there is nothing to do about it.

**Pull request merged**